### PR TITLE
Fix NPE in GraphQL

### PR DESCRIPTION
**Pull request by arv**
When we have a cyclic type we ended up getting the non complete type out of the map.

Use graphql.FieldsThunk which is designed to allow recursive types.

### CODE REVIEW

**Comment by arv** [comment]
I need to update the comment about how we use NonNull to support cycles...

**Comment by arv** [comment]
Updated comment and removed hack.

File: go/ngql/query_test.go [hunk 1]
```diff
@@ -280,7 +280,6 @@ func (suite *QueryGraphQLSuite) TestListOfUnionOfScalars() {
 
 	suite.assertQueryResult(list, "{root{elements{... on BooleanValue{b: scalarValue} ... on StringValue{s: scalarValue} ... on NumberValue{n: scalarValue}}}}", `{"data":{"root":{"elements":[{"n":28},{"s":"bar"},{"b":true}]}}}`)
 }
-
```
Comment (kalman) [inline go/ngql/query_test.go]: add back

**Review by kalman — APPROVED** [review]
lgtm

**Pull request merged**